backend/internal/user/handler/userhandler.go

```diff
@@ -0,0 +1,258 @@
+}
+
+// HandleUserPutRequest handles the user request.
+//
+// @Summary      Update an user
+// @Description  Update the details of an existing user.
+// @Tags         users
+// @Accept       json
+// @Produce      json
+// @Param        id           path   string            true  "User ID"
+// @Param        user  body   model.User  true  "Updated user data"
+// @Success      200  {object}  model.User
+// @Failure      400  {string}  "Bad Request: The request body is malformed or contains invalid data."
+// @Failure      404  {string}  "Not Found: The user with the specified ID does not exist."
+// @Failure      500  {string}  "Internal Server Error: An unexpected error occurred while processing the request."
+// @Router       /users/{id} [put]
+func (ah *UserHandler) HandleUserPutRequest(w http.ResponseWriter, r *http.Request) {
+
+	logger := log.GetLogger().With(log.String(log.LOGGER_KEY_COMPONENT_NAME, "UserHandler"))
+
+	id := strings.TrimPrefix(r.URL.Path, "/users/")
+	if id == "" {
+		http.Error(w, "Missing user id", http.StatusBadRequest)
+		return
+	}
+
+	var updatedUser model.User
+	if err := json.NewDecoder(r.Body).Decode(&updatedUser); err != nil {
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+	updatedUser.Id = id
+
+	// Update the user using the user service.
+	userProvider := userprovider.NewUserProvider()
+	userService := userProvider.GetUserService()
+	user, err := userService.UpdateUser(id, &updatedUser)
+	if err != nil {
+		http.Error(w, "Failed get user", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(user)
+
+	// Log the user response.
+	logger.Debug("User PUT response sent", log.String("user id", id))
+}
+
+// HandleUserDeleteRequest handles the user request.
+//
+// @Summary      Delete an user
+// @Description  Delete an user using its ID.
+// @Tags         users
+// @Accept       json
+// @Produce      json
+// @Param        id   path   string  true  "User ID"
+// @Success      204
+// @Failure      400  {string}  "Bad Request: The request body is malformed or contains invalid data."
+// @Failure      404  {string}  "Not Found: The user with the specified ID does not exist."
+// @Failure      500  {string}  "Internal Server Error: An unexpected error occurred while processing the request."
+// @Router       /users/{id} [delete]
+func (ah *UserHandler) HandleUserDeleteRequest(w http.ResponseWriter, r *http.Request) {
+
+	logger := log.GetLogger().With(log.String(log.LOGGER_KEY_COMPONENT_NAME, "UserHandler"))
+
+	id := strings.TrimPrefix(r.URL.Path, "/users/")
+	if id == "" {
+		http.Error(w, "Missing user id", http.StatusBadRequest)
+		return
+	}
+
+	// Delete the user using the user service.
+	userProvider := userprovider.NewUserProvider()
+	userService := userProvider.GetUserService()
+	err := userService.DeleteUser(id)
+	if err != nil {
+		http.Error(w, "Failed delete user", http.StatusInternalServerError)
```

Comment:
Maybe should return 200 for user non existing case

Reply:
Actually, this is the current behavior.. When deleting an user with an invalid id, it execute without and issue and returns a 204.